diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/VariableGenericTypeDeclarationRule.kt

```diff
@@ -0,0 +1,47 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_ARGUMENT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_PARAMETER
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class VariableGenericTypeDeclarationRule(private val configRules: List<RulesConfig>) : Rule("variable-generic-type") {
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when(node.elementType) {
+            PROPERTY, VALUE_PARAMETER -> handleProperty(node)
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleProperty(node: ASTNode) {
```

Comment:
What about `lazy properties`? You don't check them.

Reply:
Not sure that we need to check them, because they won't work in these cases:
        val lazyValue: List<Int> by lazy {
            emptyList()
        }
        val lazyValue: List<Int> by lazy {
            listOf()
        }
But it works in this case:
        val lazyValue: List<Int> by lazy {
            listOf(1,2,3)
        }
Do we need to check it? @petertrr

Reply:
Seems kotlin compiler isn't yet smart enough to determine type in cases you wrote. However, user can specify type of `lazy`:
`val lazyValue by lazy<List<Int>> { emptyList() }`, even though I think no one uses `lazy` like this. But I think these statements are of type `PROPERTY` too, aren't they?

Reply:
Yes, but first it has PROPERTY_DELEGATE child and only then call expression. So should we consider cases described above?

Reply:
No, we can't reliably determine type of the property for arbitrary delegate, and we can leave it as is for now.